token-lending/program/src/processor.rs

```diff
@@ -982,6 +983,18 @@ fn process_refresh_obligation(program_id: &Pubkey, accounts: &[AccountInfo]) ->
         }
 
         liquidity.accrue_interest(borrow_reserve.liquidity.cumulative_borrow_rate_wads)?;
+        max_borrow_weight = match max_borrow_weight {
+            None => Some((borrow_reserve.borrow_weight(), index)),
+            Some((cur_max_weight, _)) => {
```

Comment:
stable order by 2nd order by pubkey (maybe in future we could use reserve util desc or total borrow value desc)

also maybe investigate if using borrow_weight() vs config.additional_borrow_weight makes a big difference in compute usage.